string mode_req
string mode_req2
---
string mode_res
